### ISSUE TITLE
Feedback for "User Groups"

### ISSUE THREAD

**Issue by piejanssens** [SAP-docs/btp-identity-authentication#16]
https://help.sap.com/docs/IDENTITY_AUTHENTICATION/6d6d63354d1242d185ab4830fc04feb1/ddd067c899f94e2f9006cc4dd417be80.html

I'm looking for documentation about how IPS jobs can affect user group removal.

Cfr. https://answers.sap.com/questions/13847171/when-are-ias-user-groups-assignments-automatically.html
Not sure why it got downvoted, but I'm still looking for information on this.

Best regards,

Pieter

**Comment by ValAta** [comment]
Hi Pieter,
This is the [documentation ](https://help.sap.com/docs/IDENTITY_PROVISIONING/f48e822d6d484fa5ade7dda78b64d9f5/668c99120f244df19fd63b8788409b48.html?version=Cloud) for the Resync Job you were looking for. 
Does it answer your question?
BR,
Valentin

**Comment by piejanssens** [comment]
Hi Valentin,

It does, yes. Thank you. 

IAS should make a distinction between manually assigned groups and IPS assigned groups. 
Consequently, it would be great to have the option to keep manually assigned groups when executing "Resync Job".

Pieter

**Comment by ValAta** [comment]
Hi Pieter,
You are welcome!
As for your feedback, could you give more details about your scenario? What kind of problem would be solved with the option to keep manually assigned groups when executing "Resync Job"?
BR,
Valentin

**Comment by piejanssens** [comment]
SAP Work Zone requires admins to be of a certain IAS group (Workzone_Admin).

Instead of adding this authorization-only differentiator in a dummy attribute/property somewhere in the source system, it's easier to maintain these admin roles directly in IAS. A.t.m., the Work Zone admins loose their permissions as soon as anyone runs the "Resync Job". 

Without the option to keep the group assignments when executing "Resync Job", directly editing user groups could only be useful/safe if IAS is used without it being target system of IPS. IAS is a 'special' target in that sense, since it's used as a user store (including group assignments) for the SSO flows (which is not the case for other targets).

**Comment by piejanssens** [comment]
Hi Valentin,

Interesting. In this case I'm in the "manage the group only in the admin console" scenario.
I see that I have a leftover hardcoded user condition to assign the same group from the initial setup...

Are you saying that for a "Resync Job", IPS only considers deleting the user group assignments for those groups that are part in the transformation for that target system? If so, that would be great and the only thing possibly missing is this being documented.

Best regards,

Pieter

**Comment by IvelinaKiryakova** [comment]
Hi Pieter,

My name is Ivelina Kiryakova. I’m covering the IPS documentation.
If I got it right, your provisioning scenario includes replicating users and groups from SAP Work Zone, standard edition (source system) to IAS (target system). You have Workzone_Admin group in IAS. Also, every time you run a Resync job, the assigned users to the Workzone_Admin group in IAS lose their assignments.

You may probably know that a [Resync job](https://help.sap.com/docs/IDENTITY_PROVISIONING/f48e822d6d484fa5ade7dda78b64d9f5/668c99120f244df19fd63b8788409b48.html?version=Cloud) makes a full replace of users and groups in the target system with users and groups from the source system. Normally, it is used to fix inconsistent data between both systems. Could this be the reason for losing the assignments?

Can you test one of the IPS recently released functionality described in [Enabling Group Assignment](https://help.sap.com/docs/IDENTITY_PROVISIONING/f48e822d6d484fa5ade7dda78b64d9f5/0d80033336474468bb64ef8aeb7e3dd8.html?version=Cloud) and the following [blog](https://blogs.sap.com/2023/03/16/group-assignments-based-on-user-attributes-a-flexible-solution-for-managing-conditional-risk-based-authentication-and-much-more/).
Basically, in the target system transformation, you need to define a condition to filter out the Work Zone administrators, add the ID of the Workzone_Admin group and specify the targetVariable for assignGroup as explained in the documentation.

Hope this helps.
Let us know if it works.

Best regards,
Ivelina

**Comment by piejanssens** [comment]
Hi @IvelinaKiryakova,

Yes, I do think that the group assignments are deleted due to the "Resync Job". 

> You want to manage the group only in the admin console. If this is the case, you must ensure that this group is not read/mapped via the Identity Provisioning jobs

With this info, Valentin was giving me some hope that the group assignment deletions would be limited to those groups that are part of the transformation of the said target system (IAS in this case). So this is not the case after all?

The group assignment solution would require us to maintain the transformation to maintain the admin list. I was hoping that there was an option to use the IAS UI.
